container-core/src/main/java/com/yahoo/processing/request/CloneHelper.java

```diff
@@ -98,7 +98,7 @@ else if (object instanceof HashSet)
         try {
             return cloneByReflection(object);
         } catch (IllegalArgumentException e) {
-            if ( ! (e.getCause() instanceof ClassCastException))
+            if ( ! (e.getCause() != null && ClassCastException.class.getName().equals(e.getCause().getClass().getName())))
```

Comment:
Log exception before `if` with level `FINE` and enable verbose logging for this class through feature flag.